AdminUi/apps/admin_ui/lib/components/shared/date_filter.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:admin_ui/core/constants.dart';
+import 'package:flutter/material.dart';
+
+class DateFilter extends StatefulWidget {
+  const DateFilter({
+    required this.operators,
+    required this.onDateSelected,
+    required this.label,
+    super.key,
+  });
+
+  final void Function(DateTime selectedDate, String operator, {bool isDateSelected}) onDateSelected;
+  final List<String> operators;
+  final String label;
+
+  @override
+  State<DateFilter> createState() => _DateFilterState();
+}
+
+class _DateFilterState extends State<DateFilter> {
+  late String operator = '=';
+  late DateTime selectedDate = DateTime.now();
+  late bool isDateSelected = false;
```

Comment:
can all be private.

and why are these late? these felds are defined from the start, so no late required.